### PR TITLE
Added code examples for description list

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-layout?_sp.spa=pr-description-list-code-examples](https://developer.blackbaud.com/skyux-layout?_sp.spa=pr-description-list-code-examples)

File: src/app/docs/description-list/description-list-docs.component.html [hunk 1]
```diff
@@ -81,4 +81,20 @@
...
+    <sky-docs-code-example
+      heading="Vertical mode"
+      sourceCodePath="src/app/public/plugin-resources/code-examples/description-list/vertical/"
+    ></sky-docs-code-example>
+    <sky-docs-code-example
+      heading="Horizontal mode"
+      sourceCodePath="src/app/public/plugin-resources/code-examples/description-list/horizontal/"
+    ></sky-docs-code-example>
+    <sky-docs-code-example
+      heading="Long description mode"
```
Comment (blackbaud-johnly) [inline src/app/docs/description-list/description-list-docs.component.html:95]:
```suggestion
      heading="Long-description mode"
```

File: src/app/public/plugin-resources/code-examples/description-list/horizontal/description-list-demo.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+@Component({
+  selector: 'app-description-list-demo',
+  templateUrl: './description-list-demo.component.html'
+})
+export class DescriptionListDemoComponent {
+
+  public items: { term: string, description: string }[] = [
+    {
+      term: 'College',
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/description-list/horizontal/description-list-demo.component.ts:13]: Should either the horizontal or vertical examples include an example of a description that includes an unordered list? I didn't realize this was an option until reviewing the Design guidelines. Is it worth demonstrating in the code sample? Or is it simple enough that we don't need to demonstrate it?

**Review by blackbaud-johnly — COMMENTED** [review]
One suggested doc tweak and one question.

File: src/app/public/plugin-resources/code-examples/description-list/horizontal/description-list-demo.component.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/description-list/horizontal/description-list-demo.component.ts:13]: The unordered lists are separate. We'll have some styles for those soon. I don't think they wouldn't belong in these examples.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-layout?_sp.spa=pr-description-list-code-examples](https://developer.blackbaud.com/skyux-layout?_sp.spa=pr-description-list-code-examples)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-layout?_sp.spa=pr-description-list-code-examples](https://developer.blackbaud.com/skyux-layout?_sp.spa=pr-description-list-code-examples)

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**